Bugfix/next css redirect issue

ColinKrist opened the pull request:
After Mitchell added ag-grid to the homepage we have had major issues with redirection after login. This PR fixes that by applying a hacky fix for next-css css imports by adding a blank css file to the _app.jsx page. This apparently does something to fix the css import resolver and fixes Router.push not being able to redirect correctly...

I've also added a notification on the login page that listens to Router and will output an error if the .push event for redirecting to / on login fails. It tells them to close and re-open their browser. This may or may not work. But will give Kelly a good idea what might be going on.
![image](https://user-images.githubusercontent.com/27971797/56102459-d929c280-5ef2-11e9-9b6f-2993326d979d.png)

ColinKrist committed:
next-css plugin has issue with Router.push not working for pages that have css imports. this fixes that

ColinKrist committed:
remove path from cookies because it's not used and is defaulted to /

ColinKrist committed:
add notificaiton bar to login page and show err to user when login redirect fails

MitchellSlavik commented on zoo_frontend/src/util/withAuth.jsx:
```diff
@@ -41,17 +39,18 @@ export default (allowedRoles = ['authenticated']) => (WrappedComponent) => {
+          // console.log('pass validation');
+        }, () => {
           if (process.browser) {
-            document.cookie = 'authToken=; path=/';
```
I am fairly certain these still need the path because if we set token on something that is not the root path then it will not apply to all routes.

ColinKrist replied:
I wasn't facing that issue at all. the / path is set by default. I was navigating just fine and verifying that I wasn't calling login or anything and that there were no issues with the cookie. Adding back in to please.

ColinKrist committed:
adding path back in

buffginger reviewed (approved):
Love it

MitchellSlavik commented on zoo_frontend/src/util/withAuth.jsx:
```diff
@@ -41,17 +39,18 @@ export default (allowedRoles = ['authenticated']) => (WrappedComponent) => {
           api.setToken('');
-          return {};
+          redirectTo('/login', { res: ctx.res, status: 301 });
+          return { ...pageProps };
```
I think this is the part that it is messing up. Returning from here does nothing. To do the return correctly you would have to use a try/catch

ColinKrist committed:
buncha fixes

ColinKrist commented:
Per mitchell's request I've removed the .catch on validateToken to make the try catch clearer. I've also added an event listener for all router change successes and errors to hopefully make it clearer from a console standpoint why some redirects might be happening

This workflow here happens when a user's session cookie isn't present after another page transition (ie expired, blocked via adblocker)
![image](https://user-images.githubusercontent.com/27971797/56163618-64a76000-5f94-11e9-9cd9-634ce5c74fe3.png)

There were some other fixes dealing with the user login not having a role map. I've added a pop up for that too
![image](https://user-images.githubusercontent.com/27971797/56166281-d71b3e80-5f9a-11e9-94ce-a5c9d4d2e08f.png)

MitchellSlavik commented on zoo_frontend/src/pages/login/login.jsx:
```diff
@@ -60,8 +64,25 @@ class Login extends Component {
+      }
+      return;
     } catch (err) {
+      console.log(err.message);
```
nit: remove this console statement err is already being printed out

MitchellSlavik commented on zoo_frontend/src/pages/login/login.jsx:
```diff
@@ -60,8 +64,25 @@ class Login extends Component {
     } catch (err) {
+      console.log(err.message);
+      if (err.message === 'Role not found') {
+        setTimeout(() => {
```
is this settimeout necessary?

ColinKrist replied:
sadly yeah, something is going on with the timing of this that I couldn't get the notification to fire properly unless it wasn't in this sequence. Maybe cause it's in a catch block? idk

MitchellSlavik commented on zoo_frontend/src/util/withAuth.jsx:
```diff
@@ -19,37 +19,40 @@ const redirectTo = (destination, { res, status } = {}) => {
-          }
-          api.setToken('');
-          return {};
+        await api.validateToken().then(async () => {
```
nit, unneeded 'then'

ColinKrist committed:
feedback
